website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,100 @@
+
+If you need to use the state from another provider in a side-effect, you can use Ref's [read] method to obtain the current state from another provider.
+
+Typically when constructing a `Notifier` class you should pass in the `ref` to allow the Notifier to obtain the current value of dependencies as needed.
+
+### Alive -> Paused
+When an **Alive** provider is no longer listened to by other providers or the UI, it goes into a **Paused** state. 
+This means that it no longer will react to changes on providers it is listening to.
+This is an optimization, as if you are not listening to the provider, there is no need to keep it alive. 
+Every provider not being used will be returned to a **Paused** state, reducing the computational burden of your app.
+
+If you need to keep a provider alive for side-effects, make sure to listen to it in an appropriate place in the UI where it should be kept alive.
+
+If you need to perform some action when a provider is paused use the ref's [onCancel] method to register callbacks.
+
+If you need to perform some action when a provider resumes to an Alive state from a paused state, use the ref's [onResume] method to register callbacks.
+
+If you want the state to be disposed, so that in addition to taking no computational resources, it also disposes of the memory of the state, use the `.autoDispose` modifier on your provider definition.
+This will cause it to transition to a **Disposed** state instead of **Paused** when it is no longer being used.
+
+### Alive -> Disposing
+
+There are a few reasons for a provider to be disposed.
+
+- When defined using the `.autoDispose` modifier and no longer being watched by the UI or another provider
+- When the provider is being manually refreshed or invalidated
+- When the provider is being recreated due to one of it's watched dependencies changing
+
+Refreshing causes the provider to immediately go through the creation process again, whereas invalidating causes the next read / watch of the provider to cause the provider to be rebuilt.
+
+## Performing actions before the state destruction
+If you need to perform some action when a provider is disposed, use the ref's [onDispose] method to register callbacks.
+
+The following example uses onDispose to close a StreamController:
+
+<CodeBlock>{trimSnippet(onDispose)}</CodeBlock>
+
+:::note
+Depending on the provider used, it may already take care of the clean-up process.
+For example, [StateNotifierProvider] will call the `dispose` method of the returned [StateNotifier].
+:::
+
+[onDispose]: https://pub.dev/documentation/riverpod/latest/riverpod/Ref/onDispose.html
+[listen]: https://pub.dev/documentation/riverpod/latest/riverpod/Ref/listen.html
+[onCancel]: https://pub.dev/documentation/riverpod/latest/riverpod/Ref/onCancel.html
+[onResume]: https://pub.dev/documentation/riverpod/latest/riverpod/Ref/onResume.html
```

Comment:
onCancel / onResume are not in the 'latest' documentation. Is there a way to get links to the dev documentation, or should I remove the links until then?